### PR TITLE
TE-9639 Fix exception during guest checkout

### PR DESCRIPTION
**Pull request by dmytro-dymarchuk**
Ticket: https://spryker.atlassian.net/browse/TE-7904

### CODE REVIEW

File: src/SprykerEco/Yves/Payone/Form/DataProvider/InstantOnlineTransferDataProvider.php [hunk 1]
```diff
@@ -9,13 +9,26 @@
...
 use Spryker\Yves\StepEngine\Dependency\Form\StepEngineFormDataProviderInterface;
+use SprykerEco\Yves\Payone\Dependency\Client\PayoneToStoreClientInterface;
 use SprykerEco\Yves\Payone\Form\InstantOnlineTransferSubForm;
 
 class InstantOnlineTransferDataProvider implements StepEngineFormDataProviderInterface
 {
+    /**
+     * @var \SprykerEco\Yves\Payone\Dependency\Client\PayoneToStoreClientInterface
+     */
+    private $storeClient;
```
Comment (olhalivitchuk) [inline src/SprykerEco/Yves/Payone/Form/DataProvider/InstantOnlineTransferDataProvider.php:22]: shouldn't we use protected?

File: src/SprykerEco/Yves/Payone/Form/DirectDebitSubForm.php [hunk 2]
```diff
@@ -89,7 +88,9 @@ public function buildForm(FormBuilderInterface $builder, array $options)
         $this->addIBAN($builder)
             ->addBIC($builder);
 
-        if (Store::getInstance()->getCurrentCountry() === 'DE') {
+        $currentCountry = $options[DirectDebitSubForm::OPTION_BANK_COUNTRIES];
+
+        if ($currentCountry === 'DE') {
```
Comment (olhalivitchuk) [inline src/SprykerEco/Yves/Payone/Form/DirectDebitSubForm.php:93]: is it better to use something like protected const STORE_NAME = 'DE'; ?

File: src/SprykerEco/Zed/Payone/Business/Payment/MethodMapper/SecurityInvoice.php [hunk 3]
```diff
@@ -243,30 +237,12 @@ public function mapPaymentToRefund(SpyPaymentPayone $paymentEntity): RefundConta
...
-     *
-     * @return \SprykerEco\Zed\Payone\Business\Api\Request\Container\ContainerInterface
+     * @return \SprykerEco\Zed\Payone\Business\Api\Request\Container\Authorization\PaymentMethod\SecurityInvoiceContainer
      */
-    protected function createPaymentMethodContainerFromPayment(SpyPaymentPayone $paymentEntity): ContainerInterface
+    protected function createPaymentMethodContainerFromPayment(): SecurityInvoiceContainer
     {
         $paymentMethodContainer = new SecurityInvoiceContainer();
 
         return $paymentMethodContainer;
```
Comment (olhalivitchuk) [inline src/SprykerEco/Zed/Payone/Business/Payment/MethodMapper/SecurityInvoice.php:246]: I guess we can avoid here of creating a variable

File: src/SprykerEco/Zed/Payone/Business/PayoneBusinessFactory.php [hunk 4]
```diff
@@ -812,37 +794,50 @@ public function createDebitResponseMapper(): DebitResponseMapperInterface
...
+            PayoneApiConstants::PAYMENT_METHOD_PREPAYMENT => $this->createPrepayment(),
+            PayoneApiConstants::PAYMENT_METHOD_DIRECT_DEBIT => $this->createDirectDebit(),
+            PayoneApiConstants::PAYMENT_METHOD_PAYPAL_EXPRESS_CHECKOUT => $this->createGenericPayment(),
+            PayoneApiConstants::PAYMENT_METHOD_CASH_ON_DELIVERY => $this->createCashOnDelivery($this->getGlossaryFacade()),
+            PayoneApiConstants::PAYMENT_METHOD_KLARNA => $this->createKlarnaPaymentMapper(),
         ];
     }
 
     /**
+     * @deprecated
```
Comment (olhalivitchuk) [inline src/SprykerEco/Zed/Payone/Business/PayoneBusinessFactory.php:812]: not informative

File: src/SprykerEco/Zed/Payone/Dependency/Facade/PayoneToStoreFacadeBridge.php [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+ */
+
+namespace SprykerEco\Zed\Payone\Dependency\Facade;
+
+class PayoneToStoreFacadeBridge implements PayoneToStoreFacadeInterface
+{
+    /**
+     * @var \Spryker\Zed\Store\Business\StoreFacadeInterface
+     */
+    private $storeFacade;
```
Comment (olhalivitchuk) [inline src/SprykerEco/Zed/Payone/Dependency/Facade/PayoneToStoreFacadeBridge.php:15]: the same

File: src/SprykerEco/Zed/Payone/PayoneConfig.php [hunk 6]
```diff
@@ -98,7 +97,7 @@ public function getEmptySequenceNumber()
      *
      * @return \Generated\Shared\Transfer\PayoneStandardParameterTransfer
      */
-    public function getRequestStandardParameter()
+    public function getRequestStandardParameter(string $currencyIsoCode, string $countryIsoCode)
```
Comment (olhalivitchuk) [inline src/SprykerEco/Zed/Payone/PayoneConfig.php:100]: return type

File: src/SprykerEco/Zed/Payone/PayoneDependencyProvider.php [hunk 7]
```diff
@@ -132,6 +135,8 @@ protected function addGlossaryFacade(Container $container): Container
     }
 
     /**
+     * @deprecated
```
Comment (olhalivitchuk) [inline src/SprykerEco/Zed/Payone/PayoneDependencyProvider.php:138]: not informative

File: src/SprykerEco/Yves/Payone/Form/DataProvider/DirectDebitDataProvider.php [hunk 8]
```diff
@@ -51,9 +64,9 @@ public function getOptions(AbstractTransfer $quoteTransfer)
      */
     protected function getBankCountries()
```
Comment (olhalivitchuk) [inline src/SprykerEco/Yves/Payone/Form/DataProvider/DirectDebitDataProvider.php:65]: return type is missed

**Review by olhalivitchuk — CHANGES_REQUESTED** [review]
some small comments

File: src/SprykerEco/Zed/Payone/Dependency/Facade/PayoneToStoreFacadeBridge.php [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+     */
+    public function __construct($storeFacade)
+    {
+        $this->storeFacade = $storeFacade;
+    }
+
+    /**
+     * @return \Generated\Shared\Transfer\StoreTransfer
+     */
+    public function getCurrentStore()
```
[the remark this answers is not in the document]
Comment (dmytro-dymarchuk) [reply src/SprykerEco/Zed/Payone/Dependency/Facade/PayoneToStoreFacadeBridge.php:28]: Method signature in a bridge must be the same as an original facade.

File: src/SprykerEco/Zed/Payone/Dependency/Facade/PayoneToStoreFacadeInterface.php [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+ */
+
+namespace SprykerEco\Zed\Payone\Dependency\Facade;
+
+interface PayoneToStoreFacadeInterface
+{
+    /**
+     * @return \Generated\Shared\Transfer\StoreTransfer
+     */
+    public function getCurrentStore();
```
[the remark this answers is not in the document]
Comment (dmytro-dymarchuk) [reply src/SprykerEco/Zed/Payone/Dependency/Facade/PayoneToStoreFacadeInterface.php:15]: Method signature in a bridge must be the same as an original facade.

File: src/SprykerEco/Zed/Payone/Business/Payment/MethodMapper/Prepayment.php [hunk 11]
```diff
@@ -146,30 +145,12 @@ public function mapPaymentToRefund(SpyPaymentPayone $paymentEntity): RefundConta
...
-     *
-     * @return \SprykerEco\Zed\Payone\Business\Api\Request\Container\Authorization\PaymentMethod\CreditCardPseudoContainer
+     * @return \SprykerEco\Zed\Payone\Business\Api\Request\Container\Authorization\PaymentMethod\PrepaymentContainer
      */
-    protected function createPaymentMethodContainerFromPayment(SpyPaymentPayone $paymentEntity)
+    protected function createPaymentMethodContainerFromPayment()
     {
         $paymentMethodContainer = new PrepaymentContainer();
 
         return $paymentMethodContainer;
```
Comment (olhalivitchuk) [inline src/SprykerEco/Zed/Payone/Business/Payment/MethodMapper/Prepayment.php:154]:
```suggestion
        return $paymentMethodContainer;
```

**Review by olhalivitchuk — APPROVED** [review]
approved with one small comment

**Review by geega — APPROVED** [review]

**Pull request merged**